### DISCUSSION TITLE
MacOS Apps not work on other Mac's

### DISCUSSION THREAD

**Discussion by efec89** [airsdk/Adobe-Runtime-Support#955] [Q&A] [answered]
I'm creating macos application with adobe air.
It creates .app file. But only work my computer.
When try to run other mac devices, it says this app is broken.

**Comment by ajwfrost** [comment, marked as answer]
How are you transferring it from one computer to another? ".app" is a folder, and you'd need to zip this up properly with symbolic links being preserved and no additional folders being injected, in order for it to work properly on another device..

thanks

**Reply by efec89** [reply]
I'm creating zip file with .app file, and transfer this zip file via wetransfer.com. When try to unzip and open .app file, it says this app is broken.

**Reply by ajwfrost** [reply]
There could be a few issues with it then... I don't think a normal zip and unzip operation will keep the links set up properly, so you may end up with files instead of symbolic links to files. Plus zip sometimes adds weird __macos folders or similar. Finally: if your other computer is macos 10.5 or later, then it will have had a quarantine flag attached to it when you downloaded it: to remove this, you can go into a terminal window, `cd` to the folder that the .app is in, and run:
```
xattr -r -d com.apple.quarantine myapplication.app
```
That's probably the thing to do first! But you can also compare the two apps (on each machine) via e.g.
```
find myapplication.app -type f -exec ls -la {} \;
```
which should list all the files with details including if they have extended attributes and whether they're actually links....

thanks

**Reply by efec89** [reply]
thank you i will try now. 
I'm signing with self-signed sertificate, I cant sign with Apple Developer Distribution certificate, it says like on attachment. Do you know how to solve this too? Maybe I can bypass this issues with sign this certificate.

<img width="269" alt="Ekran Resmi 2021-07-06 18 04 44" src="https://user-images.githubusercontent.com/6522288/124623417-a6c66000-de84-11eb-9820-198dab3078b5.png">

**Reply by ajwfrost** [reply]
So you're just coming up against the Apple security requirements. If you're on a recent macOS build then you can only install apps that are either from the App Store or are signed by a registered Developer ID from Apple's Developer/Enterprise programs.

But if you don't sign it, you may still be able to enable this manually... What are you actually trying to generate here, a .air file? I didn't think that signing .air would give you that error, I've only seen that in conjunction with .ipa signing...

**Reply by efec89** [reply]
Apple Transporter app for to send application to AppStore, wants .pkg file. Adobe Animate with AIRSDK not export .pkg file. It exports .app, .dmg and .air files.
I want to sign .app file and then need to convert to .pkg file.

**Reply by ajwfrost** [reply]
You can create a .pkg file from a .app file, using pkgbuild. e.g.
```
sudo pkgbuild --install-location /Applications --component "MyApplication.app" --sign 'Developer ID Installer' --timestamp  "MyApplication.pkg"
```
See https://www.unix.com/man-page/osx/1/pkgbuild/
